lang/LangSource/SC_LanguageClient.cpp

```diff
@@ -174,11 +174,12 @@ void SC_LanguageClient::setCmdLine(const char* buf, size_t size)
 		if (isLibraryCompiled()) {
 			VMGlobals *g = gMainVMGlobals;
 
-			PyrString* strobj = newPyrStringN(g->gc, size, 0, true);
+			NewPyrObjectPtr ptr(g->gc, newPyrStringN(g->gc, size, 0, true));
+			PyrString* strobj = (PyrString*)ptr.get();
```

Comment:
It might be nice to have a templatized NewPyrObjectPtr that has a well-defined type, so we can avoid the cast. It would prevent simple mistakes and should be easily to implement, e.g.:
```
template <typename PyrT>
class TypedNewPyrObjectPtr {
    PyrT* get();
    PyrT* release();
    // etc....
}  
```
And then making `newPyrStringN` return the typed verison. This would allow opportunities for optimization / special casing for certain types too.

Reply:
Yes, I agree. I'd started out with a templated version in fact, but it was causing some problems, and as everything is a PyrObjectHdr anyway I simplified to this initially. (Sorry C++ gives me a headache sometimes, templates, esp...) But would be useful to ensure type correctness with PyrStrings and PyrArrays when using relevant functions.